app/models/media.rb

```diff
@@ -275,6 +276,37 @@ def try_https
     end
   end
 
+  def remove_parser_specific_parameters
+    parser_class = self.class.find_parser_class(self.url)
+    return unless parser_class&.respond_to?(:urls_parameters_to_remove)
+
+    params_to_remove = parser_class.urls_parameters_to_remove
+    return unless params_to_remove.any? { |param| self.url.include?(param) }
+
+    uri = URI.parse(self.url)
+    query_params = URI.decode_www_form(uri.query || '').to_h
+
+    params_to_remove.each do |param|
+      query_params.keys.each do |key|
+        query_params.delete(key) if key.include?(param)
```

Comment:
Does it mean that partial matches can happen? If that's the case, I suggest we just do strict matches here.